### PR TITLE
Add nsight support for automated debugging

### PR DESCRIPTION
**Pull request by gonidelis**
## Description

This adds Nsight support for automated (without terminal) debugging through our DevContainers. The users should use the CMake Tools VS Code extension (added) to configure and build the default specified target in Debug build mode (`all-dev-debug` CMake Preset added). Then by using the provided debugging option from the left sidebar on VS Code they can set breakpoints in their sources through the UI and step into kernel sources.

**Note**: Only use the left sidebar to initiate debugging. Don't use the bottom line bar with the "bug" button as this will not enable the Nsight extension which enables the UI overview of the CUDA environment (current thread, block etc.).

References:
[Nsight Extension](
https://developer.nvidia.com/nsight-visual-studio-code-edition)

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "(cuda-gdb) Launch",
+            "type": "cuda-gdb",
+            "request": "launch",
+            "program": "${command:cmake.launchTargetPath}",
+            "stopAtEntry": true,
+            "cwd": "${workspaceFolder}",
+            "preLaunchTask": "build debug",
```
Comment (gevtushenko) [inline .vscode/launch.json:11]: important: we don't need this

File: .vscode/tasks.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      "args": [
+          "-c",
+          "${command:cmake.tasksBuildCommand}"
+      ],
+      "options": {
+          "cwd": "${command:cmake.buildDirectory}"
+      }
+    }
+  ]
+}
```
Comment (gevtushenko) [inline .vscode/tasks.json:17]: important: we don't need this file

File: CMakePresets.json [hunk 3]
```diff
@@ -271,6 +271,17 @@
...
+    },
+    {
+      "name": "all-dev-debug",
+      "displayName": "all-dev debug",
+      "inherits": "all-dev",
+      "cacheVariables":{
+        "CCCL_ENABLE_BENCHMARKS": false,
+        "CMAKE_BUILD_TYPE": "Debug",
+        "CMAKE_CUDA_FLAGS": "-G",
+        "CMAKE_CUDA_ARCHITECTURES": "native"
```
Comment (gevtushenko) [inline CMakePresets.json:283]: question: does 'native' work for us?

File: .devcontainer/cuda12.4-llvm16/devcontainer.json [hunk 4]
```diff
@@ -31,7 +31,9 @@
     "vscode": {
       "extensions": [
         "llvm-vs-code-extensions.vscode-clangd",
-        "xaver.clang-format"
+        "xaver.clang-format",
+        "nvidia.nsight-vscode-edition",
+        "ms-vscode.cmake-tools"
```
Comment (jrhemstad) [inline .devcontainer/cuda12.4-llvm16/devcontainer.json:36]: question: What does `ms-vscode.cmake-tools` do for us?

File: CMakePresets.json (hunk 3 above)
Comment (gonidelis) [reply CMakePresets.json:283]: What do you mean us?

Comment (gevtushenko) [reply CMakePresets.json:283]: > What do you mean us?

If I set "native" as `CMAKE_CUDA_ARCHITECTURES`, will CCCL project be properly configured and built?

File: .devcontainer/cuda12.4-llvm16/devcontainer.json (hunk 4 above)
Comment (gonidelis) [reply .devcontainer/cuda12.4-llvm16/devcontainer.json:36]: As we are trying to make the process as user-friendly and seamless as possible the extension allows us to configure with the desired preset (all-dev-debug in our case), build and run from the UI so the user doesn't have to use the terminal.

Comment (jrhemstad) [reply .devcontainer/cuda12.4-llvm16/devcontainer.json:36]: Cool! I look forward to seeing a demo!

File: CMakePresets.json (hunk 3 above)
Comment (miscco) [reply CMakePresets.json]: I recently did update libcu++ to be able to build it against native AFAIK

Comment (gonidelis) [reply CMakePresets.json]: we still got cub and thrust tho, so let's keep it hardcoded?

**Review by griwes — APPROVED** [review]

**Pull request merged**